script/app.js

```diff
@@ -102,3 +125,14 @@ const deleteTodo = (taskId) => {
         renderTodos();
     }
 };
+
+const completeTodo = (taskId) => {
+    const task = todos.find((task) => task.id === taskId);
+
+    if (task && !task.isCompleted) {
+        task.isCompleted = true;
```

Comment:
without affecting the original task of todos list in the array, let's create a shallow copy first, update it and then update the todos array.